source/common/json/json_sanitizer.cc

```diff
@@ -65,34 +66,25 @@ absl::string_view sanitize(std::string& buffer, absl::string_view str) {
   if (need_slow == 0) {
     return str; // Fast path, should be executed most of the time.
   }
-  TRY_ASSERT_MAIN_THREAD {
-    // The Nlohmann JSON library supports serialization and is not too slow. A
-    // hand-rolled sanitizer can be a little over 2x faster at the cost of added
-    // production complexity. The main drawback is that this code cannot be used
-    // in the data plane as it throws exceptions. Should this become an issue,
-    // #20428 can be revived which is faster and doesn't throw exceptions, but
-    // adds complexity to the production code base.
+
+  // Slow path.
+
+  // If the input is valid utf-8, we can serialize it with Nlohmann.
+  if (utf8_range::IsStructurallyValid(str)) {
```

Comment:
I think we probably should add the right dependency for this in the BUILD file instead of picking up transitively.